### DISCUSSION TITLE
How can I detect URL changes when using Partials?

### DISCUSSION THREAD

**Discussion by dn1t** [freshframework/fresh#2233] [Q&A] [answered]
Hi, I'm using Fresh with Partials enabled. I need a way to detect url changes when using Partials, but couldn't find a right one. I tried listening for popstate events, added anchors of each pages and used MutationObserver to observe the DOM changes and check the data-current and data-ancestor, etc.

**Comment by marvinhagemeister** [comment, marked as answer]
There is not way to detect URL changes easily at the moment. It's something I'm hoping to add at some point in the future. Let's track this feature request in https://github.com/denoland/fresh/issues/2234
